test/gtest/CMakeLists.txt

```diff
@@ -46,7 +44,29 @@ foreach(SOURCE ${SOURCES})
     list(REMOVE_ITEM TESTS ${CMAKE_CURRENT_SOURCE_DIR}/${SOURCE})
 endforeach()
 
-foreach(TEST ${TESTS})
-    get_filename_component(BASE_NAME ${TEST} NAME_WE)
-    add_gtest(${BASE_NAME})
+foreach(SOURCE ${SKIP_TESTS})
+    list(REMOVE_ITEM TESTS ${CMAKE_CURRENT_SOURCE_DIR}/${SOURCE})
 endforeach()
+
+if( DISCRETE_GTEST )
```

Comment:
Is this flag "ON" by default or "OFF"?
IMO, this should be "ON" by default, or better yet, we rename the flag to something like "PACKAGE_ALL_GTESTS" and make it off by default.

Reply:
Undefined maps to false. So default is false here.

Reply:
Please check the PR description for updated info on this.